discretize/TreeMesh.py

```diff
@@ -1408,6 +1574,285 @@ def faceDiv(self):
             self._faceDiv = utils.sdiag(1.0/VOL)*D*utils.sdiag(S)*R
         return self._faceDiv
 
+    @property
+    def cellGradStencil(self):
+        if getattr(self, '_cellGradStencil', None) is None:
+            # self.number()
+
+            # # TODO: Preallocate!
+            # I, J, V = [], [], []
+            # PM = [-1, 1]*self.dim  # plus / minus
+
+            # # TODO total number of faces?
+            # offset = [0]*2 + [self.ntFx]*2 + [self.ntFx+self.ntFy]*2
+
+            # for ii, ind in enumerate(self._sortedCells):
+
+            #     p = self._pointer(ind)
+            #     w = self._levelWidth(p[-1])
+
+            #     if self.dim == 2:
+            #         faces = [
+            #                  self._fx2i[self._index([p[0], p[1], p[2]])],
+            #                  self._fx2i[self._index([p[0] + w, p[1], p[2]])],
+            #                  self._fy2i[self._index([p[0], p[1], p[2]])],
+            #                  self._fy2i[self._index([p[0], p[1] + w, p[2]])]
+            #                 ]
+            #     elif self.dim == 3:
+            #         faces = [
+            #                  self._fx2i[self._index([p[0], p[1],
+            #                                          p[2], p[3]])],
+            #                  self._fx2i[self._index([p[0] + w, p[1],
+            #                                          p[2], p[3]])],
+            #                  self._fy2i[self._index([p[0], p[1],
+            #                                          p[2], p[3]])],
+            #                  self._fy2i[self._index([p[0], p[1] + w,
+            #                                          p[2], p[3]])],
+            #                  self._fz2i[self._index([p[0], p[1],
+            #                                          p[2], p[3]])],
+            #                  self._fz2i[self._index([p[0], p[1],
+            #                                          p[2] + w, p[3]])]
+            #                 ]
+
+            #     for off, pm, face in zip(offset, PM, faces):
+            #         I += [ii]
+            #         J += [face + off]
+            #         V += [pm]
+
+            # D = sp.csr_matrix((V, (I, J)), shape=(self.nC, self.ntF))
+            # R = self._deflationMatrix('F')
+            # VOL = self.vol
+            # if self.dim == 2:
+            #     S = np.r_[self._areaFxFull, self._areaFyFull]
+            # elif self.dim == 3:
+            #     S = np.r_[self._areaFxFull, self._areaFyFull, self._areaFzFull]
+            # self._cellGradStencil = (D*R).T
+
+            self._cellGradStencil = sp.vstack([self._cellGradxStencil(),
```

Comment:
nice, this is cleaner. In #36, I am working on making the `cellGradxStencil` etc properties rather than methods. They really are properties, but we must have left out the `@property` decorator in the initial implementation. Don't worry about it for now, you can leave it as is, (it will be easier to fix all at once in a different pr) just a heads up.